dev-docs/hail-for-new-engineers.md

```diff
@@ -0,0 +1,189 @@
+- many Google Cloud Storage ([an object store](https://en.wikipedia.org/wiki/Object_storage)) buckets,
+- one Cloud SQL instance with a production database, ephemeral pull-request-test databases, and one
+  database per developer, and
+- logs for basically anything can be found in [GCP Logs](https://console.cloud.google.com/logs).
+
+We use a number of technologies:
+- Python is the language of choice for web applications, services, and anything user-facing,
+- Scala is the legacy language of the Hail Query compiler and run-time,
+- the JVM is the legacy target of the Hail Query compiler,
+- C++ is the aspirational language of high-performance services and the Hail Query compiler and run-time,
+- LLVM is the aspirational target of the Hail Query compiler,
+- Docker is our container image and run-time system, and
+- MySQL is our SQL database of choice.
+
+### Services Technology
+
+We almost exclusively write services in Python 3.7. We use a number of Python packages:
+- [`asyncio`](https://docs.python.org/3.7/library/asyncio.html) for concurrency which is built on
+  [coroutines](https://en.wikipedia.org/wiki/Coroutine) not threads,
+- [`aiohttp`](https://docs.aiohttp.org/en/stable/) for serving HTTPS requests (most services speak
+  HTTPS, shuffler speaks a custom TCP-based protocol), and
+- [`jinja2`](https://jinja.palletsprojects.com/en/2.11.x/) for "templating" which simply means
+  programmatically generating text files.
+
+A service is realized as:
+
+- a Docker image containing the executable code for the service,
+- a Kubernetes deployment (which defines the command to run, how much CPU is needed, what
+  environment variables are set, etc.),
+- a Kubernetes service (which defines which ports are accessible), and
+- possibly a database within our Cloud SQL instance.
+
+We call a complete, working set of all services and databases a "namespace". Every namespace
+corresponds to a Kubernetes namespace. All namespaces share one CloudSQL instance, but only have
+access to their databases.
+
+The default namespace contains our "production" services and is accessible to the outside world at
+https://hail.is, https://batch.hail.is, etc.
+
+Every developer and every pull request test job also has a namespace. Developer namespaces are
+accessible at https://internal.hail.is/DEVELOPER_USERNAME/SERVICE/ . Unlike the default namespace,
+every other namespace has exactly one database (containing all tables from each service's database).
+
+The default namespace includes three services that are hand-deployed and do not exist in other
+namespaces: gateway, internal-gateway, and router-resolver. All traffic from the Internet enters the
+cluster through the gateway which routes traffic to the appropriate namespace and service. All
+traffic from the Batch workers enters through the internal-gateway. The router-resolver assists the
+gateway in routing.
+
+## Genetics, Briefly
```

Comment:
I think the brief overview of genetics should be at the top of the document. It should transition from big picture to more specifics, especially as someone joining the team who might not have any previous knowledge about genetics or hail. Then there should be a transition between the overview of genetics and the role that hail plays in everything.